compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt

```diff
@@ -60,12 +64,62 @@ import org.w3c.dom.TouchEvent
 private val actualDensity
     get() = window.devicePixelRatio
 
+private interface ComposeWindowState {
+    fun init() {}
+    fun sizeFlow(): Flow<IntSize>
+}
+
+private class DefaultWindowState : ComposeWindowState {
+    private val channel = Channel<IntSize>(CONFLATED)
+
+    override fun init() {
+        window.addEventListener("resize", {
+            channel.trySend(getParentContainerBox())
+        })
+
+        initMediaEventListener {
+            channel.trySend(getParentContainerBox())
+        }
+
+        channel.trySend(getParentContainerBox())
+    }
+
+    private fun getParentContainerBox(): IntSize {
+        val documentElement = document.documentElement ?: return IntSize(0, 0)
+        return IntSize(documentElement.clientWidth, documentElement.clientHeight)
+    }
+
+    private fun initMediaEventListener(handler: (Double) -> Unit) {
+        val contentScale = actualDensity
+        window.matchMedia("(resolution: ${contentScale}dppx)")
+            .addEventListener("change", { evt ->
+                evt as MediaQueryListEvent
+                if (!evt.matches) {
+                    handler(contentScale)
+                }
+                initMediaEventListener(handler)
+            }, AddEventListenerOptions(capture = true, once = true))
+    }
+
+    override fun sizeFlow() = channel.receiveAsFlow()
+}
+
+private fun(suspend () -> IntSize).asComposeWindowState(): ComposeWindowState {
+    return object : ComposeWindowState {
+        override fun sizeFlow(): Flow<IntSize> = flow {
+            while (coroutineContext.isActive) {
+                emit(this@asComposeWindowState())
```

Comment:
I was a bit confused by this function initially. Could not understand what is actually emmited from here.

What if we add a companion object in the ComposeWindowState with a function for this purpose?
```kotlin
    companion object {
        fun createWithCustomSizeProvider(sizeProvider: (suspend () -> IntSize)?): ComposeWindowState? {
            if (sizeProvider == null) return null
            return object : ComposeWindowState {
                override fun sizeFlow(): Flow<IntSize> = flow {
                    while (coroutineContext.isActive) {
                        emit(sizeProvider())
                    }
                }
            }
        }
    }
```

Reply:
I agree, pushed this